File: app/src/main/java/com/niuda/a3jidi/laok/booklook/manager/DataManager.java
package com.niuda.a3jidi.laok.booklook.manager;

import android.content.Context;

import com.niuda.a3jidi.laok.booklook.model.Book;
import com.niuda.a3jidi.lib_base.base.http.RetrofitHelper;
import com.niuda.a3jidi.lib_base.base.http.RetrofitService;

import io.reactivex.Observable;

/**
 * (￣▽￣)"
 *
 * @version V1.0 :数据管理层
 * @author: Administrator
 * @date: 2017-11-11 14:07
 */

public class DataManager {
    private RetrofitService mRetrofitService;

    /**
     *  获取单例Retrofit
     * @param context
     */
    public DataManager(Context context){
        this.mRetrofitService = RetrofitHelper.getInstance(context).getService();
    }

    public Observable<Book> getSearchBook(String name, String tag, int start, int end){
        return mRetrofitService.getSearchBook(name, tag, start, end);
    }
}

File: app/src/main/java/com/niuda/a3jidi/laok/booklook/manager/DataManager.kt
package com.niuda.a3jidi.laok.booklook.manager

import android.content.Context
import com.niuda.a3jidi.lib_base.base.http.RetrofitHelper
import io.reactivex.Observable

/**
 * (￣▽￣)"
 *
 * @version V1.0 :数据管理层
 * @author: Administrator
 * @date: 2017-11-11 14:07
 */

class DataManager<T>(context: Context) {

    /**
 * 获取单例Retrofit
 * @param context
 */
    private val sHelper by lazy { RetrofitHelper(context) }


    fun getSearchBook(name: String, tag: String, start: Int, end: Int): Observable<*> {
        return sHelper.getInstance().service.getSearchBook(name, tag, start, end)
    }
}
